great_expectations/data_context/store/inline_store_backend.py

```diff
@@ -100,6 +100,10 @@ def _set(self, key: tuple[str, ...], value: Any, **kwargs: dict) -> None:
 
         if resource_type is DataContextVariableSchema.ALL_VARIABLES:
             config_commented_map_from_yaml = yaml.load(value)
+            # TODO: instead prevent these from being set??
+            # TODO: revert this before merge
+            for name in config_commented_map_from_yaml.get("fluent_datasources", {}):  # type: ignore[union-attr]
+                config_commented_map_from_yaml["datasources"].pop(name)  # type: ignore[union-attr]
```

Comment:
When we call `self._datasource_store.set(key=None, value=datasource)` this causes the `fluent` datasources to be added to the `"datasources"` config entry, which will then be validated using the old "Datasource" marshmallow schemas when we call `context.variables.save_config()`

This temporary hack is just to demonstrate the problem. 

The cleanest solution might be to also `stash` these values or simply remove fluent datasource types from `datasources` during the `_fluent_objects_stash()` call.

https://github.com/great-expectations/great_expectations/blob/2367d978b85ebec76c7e9bd81b6e562b0dc7f9c3/great_expectations/data_context/data_context_variables.py#L364-L398